cmake/list_append_keys_from.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
+#
+macro(list_append_keys_from listname source)
```

Comment:
Please convert this into a function (see #835).

Also passing the source list name as `source` unnecessarily restricts how this function can be called. I would suggest considering all `ARGN` (beside `listname`) to be appended.